pkg/upgrade/upgrade.go

```diff
@@ -104,17 +103,9 @@ func CreateDefaultDSC(ctx context.Context, cli client.Client) error {
 		},
 	}
 	err := cluster.CreateWithRetry(ctx, cli, releaseDataScienceCluster, 1) // 1 min timeout
-	switch {
-	case err == nil:
-		fmt.Printf("created DataScienceCluster resource\n")
-	case k8serr.IsAlreadyExists(err):
-		// Do not update the DSC if it already exists
-		fmt.Printf("DataScienceCluster resource already exists. It will not be updated with default DSC.\n")
-		return nil
-	default:
+	if err != nil {
 		return fmt.Errorf("failed to create DataScienceCluster custom resource: %w", err)
 	}
-
```

Comment:
on line 122:` If there exists an instance already, it patches the DSCISpec with default values`

can we  first clarify what we want: if exist then patch back to default or leave it as-is.
from code (even before this PR) looks like it is not gonna patch to default, because if one DSCI CR exist, then dont do anything regardless what is the name or if the spec is different.

Reply:
Great eye! I missed the comment. In the original patch 1ccbe053952c ("Unset Tech Preview components by default (#708)") it patched with the default values. But then f756e404678a ("Update incubation with downstream changes (#783)") changes it with the message I borrowed without changing the comment.

(Sorry, but due to squashing it's not obvious where the change came from, I could dig it to a4788f305573 ("fix(mm-monitoring): revert the code logic but set to disable as delete (#153)") but it squashed again and has only line `Retain existing DSCI values` with no description or commit reference around.)